src/client/Client.cc

```diff
@@ -425,18 +436,23 @@ void Client::dump_cache(Formatter *f)
   if (f)
     f->open_array_section("cache");
 
-  if (root)
-    dump_inode(f, root, did, true);
+  if (inode_no) {
+      auto it = inode_map.find(vinodeno_t(inode_no, CEPH_NOSNAP));
+      if (it != inode_map.end())
+          dump_inode(f, it->second, did, true);
+  } else {
+    if (root)
+      dump_inode(f, root, did, true);
 
-  // make a second pass to catch anything disconnected
-  for (ceph::unordered_map<vinodeno_t, Inode*>::iterator it = inode_map.begin();
-       it != inode_map.end();
-       ++it) {
-    if (did.count(it->second))
-      continue;
-    dump_inode(f, it->second, did, true);
+    // make a second pass to catch anything disconnected
+    for (ceph::unordered_map<vinodeno_t, Inode*>::iterator it = inode_map.begin();
```

Comment:
`for (const auto& p : inode_map)`

We try to update for loops to use range-based style in new code.